text/nodejs/0014-yarn-berry.md

```diff
@@ -0,0 +1,277 @@
+# Support for Yarn "Berry" 
+
+## Summary
+
+The Paketo Node.js Buildpack supports building apps which use
+[Yarn](https://yarnpkg.com/) as a package/project management tool. Over the
+last few years, the developers of Yarn have released new major versions (v2 &
+v3), collectively called Yarn "Berry", which represent a completely new
+direction for managing dependencies in Node.js projects. Chief among the
+changes is the introduction of [Plug 'n
+Play](https://yarnpkg.com/features/pnp), a system which removes the reliance on
+`node_modules` and can eliminate the need to run `yarn install` at all. The
+Yarn implementation buildpacks do not currently support building apps which use
+Berry.  
+
+There have been quite a number of changes in the behaviour of Yarn with the
+advent of Berry. Along with performance improvements and optimizations, many
+CLI operations and methods of configuration have been deprecated or removed,
+including those crucial to the functioning of the buildpack as it exists today. 
+
+## Proposal
+
+Yarn Berry seems to represent the direction in which the project is headed, as
+[stated by its lead
+maintainer](https://dev.to/arcanis/introducing-yarn-2-4eh1#conclusion). Given
+this, and the extent of the changes to the CLI, this RFC proposes that a `yarn-berry-install`
+buildpack be created to enable the new installation workflows.
+
+## Motivation
+
+As developers continue to adopt Yarn Berry, demand for the Node.js buildpack to
+support it has steadily increased.
+
+
+## Implementation
+
+
+### API
+
+`yarn-berry-install`
+
+Provides: `yarn_packages` OR `node_modules` Requires: `node`, during `build` and `launch`
```

Comment:
Why does this require anything at launch? At launch, this buildpack isn't really doing anything other than making sure that there are some JavaScript files available on the filesystem. I'd expect that if someone wanted to run these files, they'd need to require `node` some other way, no?

Reply:
You are correct, I've updated accordingly.